src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -636,7 +636,8 @@ private void ApplyCoercion(DependencyObject actualInstanceAlias, DependencyPrope
 				return;
 			}
 
-			if (!propertyDetails.Metadata.CoerceWhenUnchanged && Equals(previousValue, baseValue))
+			if (Equals(previousValue, baseValue) &&
+				propertyDetails.Metadata is FrameworkPropertyMetadata fpmd && (fpmd.Options & FrameworkPropertyMetadataOptions.CoerceWhenUnchanged) == 0)
```

Comment:
@ramezgerges Let's use extensions methods on Options instead of doing bitwise logic directly here (there are other methods doing this for other flags already)